compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -412,24 +412,23 @@ enum TupleArgumentsFlag {
     TupleArguments,
 }
 
-fn fatally_break_rust(tcx: TyCtxt<'_>) {
+fn fatally_break_rust(tcx: TyCtxt<'_>) -> ! {
     let dcx = tcx.sess.dcx();
-    dcx.span_bug_no_panic(
-        MultiSpan::new(),
-        "It looks like you're trying to break rust; would you like some ICE?",
-    );
-    dcx.note("the compiler expectedly panicked. this is a feature.");
-    dcx.note(
+    let mut diag =
```

Comment:
This could be a struct_span_bug I think. Just pass in the span from typeck.

Reply:
Sorry, I don't understand what "the span from typeck" is...

Reply:
`fatally_break_rust` gets called here:

https://github.com/rust-lang/rust/blob/236ac911de2f4b6210ba99e25c6c4843c9381a1a/compiler/rustc_hir_typeck/src/expr.rs#L728

There should be a relevant span to pass in from that function so that this can emitted with a primary span, rather than without one, like the error is being emitted right now.

Reply:
probably `expr.span` does the trick, since it'll highlight:

```rust
    break rust;
    ^^^^^^^^^^
```

Reply:
Ok, I have updated the code with this change.